group/assignment5/UnknownAlphabet.java

```diff
@@ -88,4 +85,128 @@
         }
         return alphabet;
     }
+
+    // ****** Challenge 2 **********
+
+    /**
+     * Checks, if a given dictionary is inconsistent
+     *
+     * @param dictionary list of strings
+     * @return minimal set of constrains that cannot be satisfied or empty Optional
+     */
+    public static Optional<Set<List<Character>>> checkInconsistency(List<String> dictionary) {
+        Map<Character, Set<Character>> contrains = new HashMap<>();
+        for (int i = 0; i < dictionary.size() - 1; i++) {
+            String w1 = dictionary.get(i);
+            String w2 = dictionary.get(i + 1);
+
+            for (int j = 0; j < Math.min(w1.length(), w2.length()); j++) {
+                if (w1.charAt(j) != w2.charAt(j)) {
+                    if (!contrains.containsKey(dictionary.get(i).charAt(j))) {
+                        // adds all the chars in the dictionary into the maps
+                        contrains.put(dictionary.get(i).charAt(j), new HashSet<>());
+                    }
+                    contrains.get(w1.charAt(j)).add(w2.charAt(j));
```

Comment:
i think you can remove the duplicate code existing in :findIncomingEdges (40-49) and checkInconsistency (97-109). I would create a generalFunction and use it. I would also use an "if statement" that will help you to fill the ingoingEdgeMap HashMap :) . Let me know if you like my opinion :*

Reply:
Thank you for comment. I agree with you I think it woul be a better approach but since these parts are for challenge 2 and Anne wrote these part (I didn't change any part of her code I just used the way she implemented since challenge 2 was needed for challenge 3) I think it is better say this comment to her and see what she thinks. I think it would be better to discuss with her before changing her part. So, I will remain this part as it is and if she thinks it is a good approach we can do the changes